### PR TITLE
Warn users when import sympy which is not installed

### LINKED ISSUE #131: Raise proper warning when sympy is not installed!
🐞 **Problem**
Sympy is an optional dependency! We should let users know about this!

🎯 **Goal**

💡 **Possible solutions**

📋  **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #<put issue number here>`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

### PR DESCRIPTION
**Pull request by rishi-s8**
Fixes #131 

@shreyasbapat 

Screenshot 2019-04-07 at 11 02 49 PM

### CODE REVIEW

**Comment by rishi-s8** [comment]
@shreyasbapat I don't understand why it is showing that ```Imports are incorrectly sorted```. I think the ```try-except``` statement reorders the imports. Any suggestions on how to solve this?

**Comment by rishi-s8** [comment]
To fix the error ```Imports are incorrectly sorted``` I installed Python-Isort in Atom and sort the imports using the package. Now the error has changed to ```would reformat /home/circleci/repo/src/einsteinpy/utils/christoffel.py```. :(

**Comment by shreyasbapat** [comment]
> Imports are incorrectly sorted

Where do you see that? I see formatting errors, that can be fixed by `black` . If you still see some import sorting errors, use `isort`.  Have a look at what commands the Circle CI has been configured to run. You will find your answers :)
https://circleci.com/gh/einsteinpy/einsteinpy/1720?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by shreyasbapat** [comment]
Perks of writing a production grade code :)

**Comment by rishi-s8** [comment]
I think due to the ```try-except``` block, the tests are missing the lines of code there. Therefore the coverage is decreased. Any idea how to fix this?

**Comment by shreyasbapat** [comment]
You will have to write a test :)

**Comment by rishi-s8** [comment]
@shreyasbapat Please review the PR. CodeCov hasn't checked again yet. But I found a strange behavior. 
When I run ```python -m pytest```, the warning about sympy isn't raised, and the test for warning of sympy passes successfully.
When I run ```python -m pytest test_christoffel_wo_sympy.py```, I get the warning which is the expected behaviour. Can you see why?

**Comment by shreyasbapat** [comment]
I will check today :)

**Comment by shreyasbapat** [comment]
@rishi-s8 Can you just remove that test? I will merge it anyways :) We don't need to do such a testing for that functionality!

**Comment by shreyasbapat** [comment]
But hey! I see your test passed in https://circleci.com/gh/rishi-s8/einsteinpy/9?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

File: src/einsteinpy/tests/test_utils/test_christoffel.py [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
-import sympy
 
 from einsteinpy.utils import christoffel
 
+try:
+    import sympy
+except ImportError:
+    import warnings
+
+    warnings.warn("sympy is needed but not installed.")
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_utils/test_christoffel.py]:
```suggestion
    warnings.warn("SymPy is required but is not installed.")
```

File: src/einsteinpy/tests/test_utils/test_christoffel_wo_sympy.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import os
+import sys
+
+import pytest
+
+
+def test_christoffel_wo_sympy():
+    sys.modules["sympy"] = None
+    from einsteinpy.utils import christoffel  # isort:skip
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_utils/test_christoffel_wo_sympy.py]: Well, what is the test doing? don't you feel that you should catch the output and compare it with the pre-existing string ?

File: src/einsteinpy/utils/christoffel.py [hunk 3]
```diff
@@ -1,5 +1,11 @@
 import numpy as np
-import sympy
+
+try:
+    import sympy
+except ImportError:
+    import warnings
+
+    warnings.warn("sympy is needed but not installed.")
```
Comment (shreyasbapat) [inline src/einsteinpy/utils/christoffel.py]: Same as above :)

**Comment by rishi-s8** [comment]
Okay I'll resolve the changes and push. :)

File: src/einsteinpy/tests/test_utils/test_christoffel_wo_sympy.py (hunk 2 above)
Comment (rishi-s8) [reply src/einsteinpy/tests/test_utils/test_christoffel_wo_sympy.py]: Removed the test

**Comment by rishi-s8** [comment]
@shreyasbapat I have made the changes requested in the review and the QCs have passed. :)

**Comment by shreyasbapat** [comment]
We will merge this even when we know the coverage is decreasing. Because we have no choice :/

**Comment by shreyasbapat** [comment]
Thanks @rishi-s8 :)

**Pull request merged**